tidy3d/plugins/design/method.py

```diff
@@ -23,150 +21,87 @@ class Method(Tidy3dBaseModel, ABC):
-                return kwarg_str
-            # just to be safe, handle the case if this does not work
-            except ValueError:
-                return f"{pt_index}_{sim_index}"
-
-        # for each point, construct the simulation inputs into a dict
-        simulations = {}
-        task_name_mappings = []
-        for i in range(num_points):
-            fn_kwargs = {key: vals[i] for key, vals in fn_args.items()}
-            sim = fn_pre(**fn_kwargs)
-            if isinstance(sim, Simulation):
-                task_name = get_task_name(pt_index=i, sim_index=None, fn_kwargs=fn_kwargs)
-                simulations[task_name] = sim
-                task_name_mappings.append([task_name])
-            elif isinstance(sim, dict):
-                task_name_mappings.append({})
-                for name, _sim in sim.items():
-                    task_name = get_task_name(pt_index=1, sim_index=name, fn_kwargs=fn_kwargs)
-                    simulations[task_name] = _sim
-                    task_name_mappings[i][name] = task_name
-            else:
-                task_name_mappings.append([])
-                for j, _sim in enumerate(sim):
-                    task_name = get_task_name(pt_index=i, sim_index=j, fn_kwargs=fn_kwargs)
-                    simulations[task_name] = _sim
-                    task_name_mappings[i].append(task_name)
+        fn_args = self.sample(parameters)
+        for arg_dict in fn_args:
+            self._force_int(arg_dict, parameters)
+        return fn_args
 
-        # run in batch
-        batch_data = self._run_batch(simulations=simulations, path_dir=path_dir, **batch_kwargs)
-        task_id_dict = batch_data.task_ids
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
+        """Defines the search algorithm (sequential)."""
 
-        # run post processing on each data
-        result = []
-        task_ids = []
-        for task_names_i in task_name_mappings:
-            if isinstance(task_names_i, dict):
-                task_ids.append([task_id_dict[task_name] for task_name in task_names_i.values()])
-                kwargs_dict = {
-                    kwarg_name: batch_data[task_name]
-                    for kwarg_name, task_name in task_names_i.items()
-                }
-                val = fn_post(**kwargs_dict)
-            else:
-                task_ids.append([task_id_dict[task_name] for task_name in task_names_i])
-                args_list = (batch_data[task_name] for task_name in task_names_i)
-                val = fn_post(*args_list)
+        # get all function inputs
+        fn_args = self._assemble_args(parameters)
 
-            result.append(val)
+        # Run user function on sampled args
+        results = self._extract_output(run_fn(fn_args), sampler=True)
 
-        return fn_args, result, task_ids, batch_data
+        # None is aux_output and opt_output
+        return fn_args, results, None, None
```

Comment:
what happened to the task_ids and batch_data?

Reply:
I think this Github has picked up this modification from the original implementation of the method. The "None"s now refer to different things.